File: src/main/java/net/tassia/event/EventListener.java
package net.tassia.event;

/**
 * An EventListener listens for an event.
 * @param <E> the event to listen for
 * @since EventAPI 1.0
 * @author Tassilo
 */
@FunctionalInterface
public interface EventListener<E extends Event> {

	/**
	 * Invoked when the event is called.
	 * @param event the event
	 */
	void onEvent(E event);

}

File: src/main/kotlin/net/tassia/event/EventListener.kt
package net.tassia.event

/**
 * A event listener is used to listen for a specific event, or events extending the specified event.
 *
 * @since EventAPI 1.0
 * @author Tassilo
 */
fun interface EventListener<T : Event> {

	/**
	 * Invoked when the given event is called.
	 *
	 * @param event the event
	 */
	fun onEvent(event: T)

}
